### ISSUE TITLE
Composite devices; such as WD19 information not "strictly" accurate

### ISSUE THREAD

**Issue by superm1** [hughsie/lvfs-website#327]
The first stable firmware for the WD19 was released: https://fwupd.org/lvfs/device/com.dell.wd198ceeeffd.firmware and I notice that it marks 
"Update is not cryptographically signed"

This is technically true for one of the payloads which is unsigned, but many of the payloads are signed.

I wonder if that table needs to have a half way check mark or something?  A yellow badge that can be used for composite firmwares?

**Comment by hughsie** [comment]
> Update is not cryptographically signed

Isn't that true tho? Surely a chain is as strong as its weakest link?

**Comment by superm1** [comment]
If this component was in a chain of the root of trust for the device sure I'd agree, but it's a peripheral component.  If you flashed a tampered payload you might have a non-functional component (running in bootloader) but you wouldn't be able to attack any other components.  At least that was the conclusion of the threat modeling exercise to look at what you could do if you ended up with persistent malware in a potentially tampered payload...

**Comment by hughsie** [comment]
> If you flashed a tampered payload you might have a non-functional component

Is that worst case? I can flash a ColorHug device that turns into a keyboard that just types "All work and no play makes Jack a dull boy..." over and over. Obviously `win+r, cmd, nc foo bar` would be worse.

**Comment by superm1** [comment]
In my mind this comes down to where you're looking at the security of - the device or the host.  From the device's perspective, I don't see a lot of opportunity for attack.

In this particular case the component isn't on the USB bus, so I don't think that particular attack described would work.  It does have control of the connected displays so could potentially tamper with the output.  From the host's perspective DP is a fairly open bus, but an attack would be no different than what a rogue monitor could do.

**Comment by hughsie** [comment]
> isn't on the USB bus, so I don't think that particular attack described would work

Right, I agree in this case the malicious firmware is much less bad. It's still not signed tho.

**Comment by superm1** [comment]
What about splitting up the page to show the details for all the components when it comes to composite firmwares?

**Comment by hughsie** [comment]
I guess visually it might be a lot of information. Rather than deduping each type I guess we could keep a count of how many sub-components have the specific attributes, although it does mean the user will stare at the "thing" on the LVFS and wonder "_well, is it signed or not signed_"

**Comment by superm1** [comment]
Visually another option can be to "fold" it up.  So if the information isn't the same for everything in the composite device, show something to the like (Yellow badge?) and then press an + or an arrow to see the details for all the stuff underneath it.

**Comment by hughsie** [comment]
I guess yellow would be okay, although I'm not sure how the code would look. We'll need this kind of "warning" state for firmware that needs a subscription to access (don't ask...) as well.
